pkg/authz/policy.go

```diff
@@ -116,8 +121,13 @@ func (authorizer *policyAuthorizer) Authorize(req *http.Request) (Authorization,
 		},
 	}
 
+	tokenValid := true
```

Comment:
Should we always start with the assumption that it's invalid and let the query tell us otherwise? It feels a little safer.

At the moment if the error from allowQuery is not nil, then The `Authorization.TokenValid` will be true. Even if the error is returned with it, we're relying on it being checked. If tokenValid defaulted to false, the error would be returned along with a TokenValid value of false.